goalgorilla/open_social
# Implement recommended search block feature

## Problem
The goal is to give the SM the ability to always show certain search results based on specific keywords. E.g. the “employee handbook” query should always show the Employee Handbook node on top as recommended result.

It’s kind of like a hardcoded list of results for search keywords.

## Solution
Create a separate taxonomy vocabulary for keywords and add there fields for each entity type. So, it will be possible to add a keyword and related results during creation. Then create a custom block that will process keywords with search keys.

## Issue tracker
https://getopensocial.atlassian.net/browse/YANG-6045

## How to test
Use these changes as a patch
Enable `social_search_recommended` module
Go to the `/admin/structure/taxonomy/manage/search_keywords/overview` page and add some keyword(s) and related entities during creation that term
Go to the all search page and search by prepared keyword
You should see "Recommended results" block at the top

## Screenshots
![image](https://user-images.githubusercontent.com/50984627/129526768-2bc3ca21-b850-4c0b-909b-b4a336489393.png)
![image](https://user-images.githubusercontent.com/50984627/129526793-214059fc-4146-4ae1-8425-70f1c4e8a8f7.png)
![image](https://user-images.githubusercontent.com/50984627/129526864-a6f651c4-077b-45e4-8660-6fc5a5b870d1.png)
![image](https://user-images.githubusercontent.com/50984627/129526938-30bb9132-b1e0-4569-a99c-20cedbc89da3.png)

## Release notes
No.

## Change Record
No.

## Translations
No.
